### DISCUSSION TITLE
Why does Pocketbase store Password based Authentication Tokens  in LocalStoage?

### DISCUSSION THREAD

**Discussion by aronggile** [pocketbase/pocketbase#248] [Q&A] [answered]
Pocketbase stores  authentication tokens in local storage.
- What is the reason behind this choice?
- Doesn't  storing JWT tokens in local storage weaken security(prone to XSS attacks for e.g) 
   and in general discouraged?

**Comment by ganigeorgiev** [comment, marked as answer]
> What is the reason behind this choice?

Because it is easier and more convenient this way.

> Doesn't storing JWT tokens in local storage weaken security(prone to XSS attacks for e.g)
and in general discouraged?

If there is a XSS vulnerability in the application an attacker will be able to access the auth token, **but that's a big "IF"**.
With an exception to some trusted fields related to the docs, all dynamic strings in the application are inserted as plain text. PocketBase doesn't load any external JS so this also greatly limit the possibilities of XSS. Further more, the files serving action has a default CSP (see https://github.com/pocketbase/pocketbase/discussions/164#discussioncomment-3196837), similar will be added in the future for the Admin UI too, but  there are some difficulties related to the generated dynamic vite bundle code that will need to get resolved first.

Suggestions for improvements are always welcome. The alternative usually is to use cookie, but that comes with its own peculiarities (see https://github.com/pocketbase/pocketbase/issues/215#issuecomment-1200470851).

**Reply by aronggile** [reply]
Okay, thank you for addressing this thoroughly. I think this is an important information and maybe 
it can be included in the main doc for better reference. Especially for those planning to extend 
Pocketbase. 

Thank you for your time!

**Reply by ganigeorgiev** [reply]
Once a default CSP is added to the Admin UI (I will try to prioritize it for v0.4.0) this will eliminate almost all XSS attack vectors. So there is no need to be added in the docs.

For user-land code, the currently available SDKs ([JavaScript](https://github.com/pocketbase/js-sdk) and [Dart](https://github.com/pocketbase/dart-sdk)) supports extending the default AuthStore so the developers are able to provide whatever custom auth managing implementation they want (the js sdk will also have a built-in optional cookie store for easier SSR integration related to this issue https://github.com/pocketbase/js-sdk/issues/15).
